OSDOCS#3630: OTA etcd backup guidance

skopacz1 opened the pull request:
[OSDOCS-3630](https://issues.redhat.com/browse/OSDOCS-3630)

Versions: 4.14+

This PR adds guidance in the OTA docs about how to utilize etcd backups in the context of updates. In its current form, the content mostly serves to tell users "don't use this as a rollback solution (unless you really, really, really need to)" and then points them towards existing content that explains more about backup and restore operations.

QE review:
- [x] QE has approved this change.

Preview: [etcd backups before cluster updates](https://71747--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/updating-cluster-prepare#update-etcd-backup_updating-cluster-prepare)

skopacz1 commented:
@JianLi-RH could you PTAL when you have a chance? Thanks!

JianLi-RH commented:
hi @skopacz1 Could you please take a look if the preview page not working? I can't see the changes on it.

JianLi-RH commented:
lgtm without a preview page.
cc etcd guys for further check. /cc @geliu2016  @sandeepknd

geliu2016 commented:
/lgtm

skopacz1 commented:
@JianLi-RH sorry, I moved the location of this section and did not update the preview link accordingly. I've updated it now to point to the correct location: [etcd backups before cluster updates](https://71747--ocpdocs-pr.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/updating-cluster-prepare#update-etcd-backup_updating-cluster-prepare)

skopacz1 commented:
/label peer-review-needed

cbippley commented:
/label peer-review-in-progress
/remove-label peer-review-needed

cbippley commented on modules/update-etcd-backup.adoc:
```diff
@@ -0,0 +1,24 @@
+etcd backups record the state of your cluster and all of its resource objects.
+You can use backups to attempt restoring the state of a cluster in disaster scenarios where you cannot recover a cluster in its currently dysfunctional state.
+
+In the context of updates, you can attempt an etcd restoration of the cluster in a case where an update introduced catastrophic conditions that cannot be fixed without reverting to the cluster version prior to the update.
```
```suggestion
In the context of updates, you can attempt an etcd restoration of the cluster in a case where an update introduced catastrophic conditions that cannot be fixed without reverting to the cluster version before the update.
```
Vale flagged this.

cbippley commented on modules/update-etcd-backup.adoc:
```diff
@@ -0,0 +1,24 @@
+If your update is failing to complete, contact Red{nbsp}Hat support.
+====
+
+There are a number of factors that affect the viability of an etcd restoration.
```
```suggestion
There are several factors that affect the viability of an etcd restoration.
```
Vale flagged this.

cbippley commented on modules/update-etcd-backup.adoc:
```diff
@@ -0,0 +1,24 @@
+etcd backups record the state of your cluster and all of its resource objects.
+You can use backups to attempt restoring the state of a cluster in disaster scenarios where you cannot recover a cluster in its currently dysfunctional state.
+
+In the context of updates, you can attempt an etcd restoration of the cluster in a case where an update introduced catastrophic conditions that cannot be fixed without reverting to the cluster version prior to the update.
```
```suggestion
In the context of updates, you can attempt an etcd restoration of the cluster in a case where an update introduced catastrophic conditions that cannot be fixed without reverting to the cluster version prior to the update.
```
I'm finding this sentence difficult to parse out. Maybe try to simplify it?

skopacz1 replied:
How about this:
```suggestion
In the context of updates, you can attempt an etcd restoration of the cluster if an update introduced catastrophic conditions that cannot be fixed without reverting to the previous cluster version.
```

cbippley replied:
Yes, that's better!

cbippley reviewed (commented):
Couple nits, otherwise LGTM!

cbippley commented:
/label peer-review-done
/remove-label peer-review-in-progress

skopacz1 commented:
/label merge-review-needed

opayne1 reviewed (commented):
/lgtm

opayne1 commented:
/cherrypick enterprise-4.14

opayne1 commented:
/cherrypick enterprise-4.15

opayne1 commented:
/cherrypick enterprise-4.16
